Adds a new section that describes WithReference and equivalent methods in the "What are Aspire Integrations?" article (#795)
microsoft/aspire.dev: src/frontend/src/content/docs/integrations/overview.mdx

Hosting and client integrations work together but are **not** coupled and can be used independently. Configuration via environment variables connects client integrations to their corresponding hosting integrations, with the AppHost project managing this connection.

## Wiring resources to consuming projects with references

> Once you've created resources, such as databases or messaging systems, by setting up hosting integrations in the AppHost, you can pass those resources to consuming projects, such as APIs or web apps, by calling the `WithReference` method, or its equivalent for your preferred language. These methods express an explicit dependency between a resource and a consuming project. When Aspire starts, it injects connection information as environment variables into the consuming project so it can locate and connect to that resource.

Reviewer: ```suggestion
Once you've created resources, such as databases or messaging systems, by setting up hosting integrations in the AppHost, you can pass those resources to consuming projects, such as APIs or web apps, by calling the "with reference" APIs. These methods express an explicit dependency between a resource and a consuming project. When Aspire starts, it injects connection information as environment variables into the consuming project so it can locate and connect to that resource.
```

Author: Yeah - this works well. Fixed!

> <PivotSelector

Reviewer: Maybe we should have a generic mermaid diagram here or nearby, the is similar to what was added in the Postgres get started, that shows two "resources" modeled in the AppHost and helps visualize how hosting and client integrations relate from that perspective.

Author: Done. I've done one for C# and one for TypeScript.

  title="Select your AppHost language"
  key="aspire-lang"
  options={[
    { id: 'csharp', title: 'C#' },
    { id: 'typescript', title: 'TypeScript' },
  ]}

> />

Reviewer: Use tabs here, no need for the pivot. You're already using tabs below.

Author: Fixed.

<Pivot id="csharp">

```csharp title="C# — AppHost.cs"
var builder = DistributedApplication.CreateBuilder(args);

var cache = builder.AddRedis("cache");
var db = builder.AddPostgres("postgres").AddDatabase("appdb");

builder.AddProject<Projects.Api>("api")
    .WithReference(cache)
    .WithReference(db);

builder.Build().Run();
```

</Pivot>

<Pivot id="typescript">

```typescript title="TypeScript — apphost.ts" twoslash
import { createBuilder } from './.modules/aspire.js';

const builder = await createBuilder();

const cache = await builder.addRedis("cache");
const db = (await builder.addPostgres("postgres")).addDatabase("appdb");

const api = await builder.addProject("api", "../Api/Api.csproj");
await api.withReference(cache);
await api.withReference(db);

await builder.build().run();
```

</Pivot>

### Environment variables injected with references

The environment variables Aspire injects depend on the type of resource being referenced. For example:

> - **Connection string resources**: Aspire sets a `ConnectionStrings__{name}` environment variable, where `{name}` matches the resource name. For example, `.WithReference(cache)` on a resource named `"cache"`, that is available on port 6379, injects `ConnectionStrings__cache=localhost:6379`.

Reviewer: Try to avoid a callout to an explicit .NET API, simply say something like "when one resource chains a call to reference another resource". Leave the code parts for the tabs.

Author: Fixed.

> - **Service endpoint resources**: Aspire sets `services__{name}__{scheme}__{index}` variables used for service discovery. For example, referencing a project named `api` injects `services__api__https__0=https://localhost:7001`.

Author: Fixed. I've left the specific port number but changed the text to indicate that the port number depends on the resource.

> - **Service endpoint resources**: Aspire sets `services__{name}__{scheme}__{index}` variables used for service discovery. For example, referencing a project named `api`, that is available on port 7001, injects `services__api__https__0=https://localhost:7001`.

Reviewer: Use the simple env var names instead of the legacy names.

Author: Fixed.

Other environment variables often include credentials, port numbers, and other connection details.

### Consuming the injected variables

Any application can read these environment variables to connect to its dependencies. A Python, Node.js, or other app can read variables directly, without any Aspire-specific libraries.

> If your consuming project is written in .NET C#, you have an additional option: [**Aspire client integrations**](#client-integrations):

Reviewer: Try to avoid saying ".NET", "C#" is much better branding. Also, I think we should just let the tabs speak for themselves. Instead of calling out C# here - instead let the reader select C# if they want those details. In other words, please just fold this into the C# tab, let the reader know they can manually use Environment.GetEnvironmentVariable or the IConfiguration APIs, or preferably use the official client integrations.

```suggestion
If your consuming project is written in C#, you have an additional option: [**Aspire client integrations**](#client-integrations):
```

Author: Fixed.

<Tabs syncKey="aspire-lang">
<TabItem id="csharp" label="C#">

```csharp title="C# — Program.cs (consuming .NET project)"

> var builder = WebApplication.CreateBuilder(args);

Author: Fixed. I've used a new sync key.
